File: php/ltx/src/InterplanetLTX/LtxSegment.php
<?php
/**
 * LtxSegment.php — Computed, timed segment with UTC epoch milliseconds
 * Story 33.4 — PHP LTX library
 */

namespace InterplanetLTX;

/** A computed, timed LTX segment. */
readonly class LtxSegment
{
    public function __construct(
        public string $type,
        public int    $q,
        public int    $startMs,
        public int    $endMs,
        public int    $durMin,
    ) {}
}

File: java/ltx/src/com/interplanet/ltx/LtxSegment.java
package com.interplanet.ltx;

/**
 * LtxSegment — a computed timed segment with absolute start/end times.
 * Story 33.2 — Java LTX library
 */
public record LtxSegment(
    /** Segment type, e.g. "TX", "RX". */
    String type,
    /** Duration in quanta. */
    int q,
    /** Absolute start time (UTC milliseconds since Unix epoch). */
    long startMs,
    /** Absolute end time (UTC milliseconds since Unix epoch). */
    long endMs,
    /** Duration in minutes (q * quantum). */
    int durMin
) {}
